IonHashDotNet/IonHashReader.cs

```diff
@@ -0,0 +1,240 @@
+            }
+
+            this.ionType = this.reader.MoveNext();
+            return this.ionType;
+        }
+
+        // implements IIonReader
+        public ISymbolTable GetSymbolTable()
+        {
+            return this.reader.GetSymbolTable();
+        }
+
+        public bool BoolValue()
+        {
+            return this.reader.BoolValue();
+        }
+
+        public BigInteger BigIntegerValue()
+        {
+            return this.reader.BigIntegerValue();
+        }
+
+        public int IntValue()
+        {
+            return this.reader.IntValue();
+        }
+
+        public long LongValue()
+        {
+            return this.reader.LongValue();
+        }
+
+        public double DoubleValue()
+        {
+            return this.reader.DoubleValue();
+        }
+
+        public BigDecimal DecimalValue()
+        {
+            return this.reader.DecimalValue();
+        }
+
+        public Timestamp TimestampValue()
+        {
+            return this.reader.TimestampValue();
+        }
+
+        public string StringValue()
+        {
+            return this.reader.StringValue();
+        }
+
+        public SymbolToken SymbolValue()
+        {
+            return this.reader.SymbolValue();
+        }
+
+        public IntegerSize GetIntegerSize()
+        {
+            return this.reader.GetIntegerSize();
+        }
+
+        public SymbolToken GetFieldNameSymbol()
+        {
+            return this.reader.GetFieldNameSymbol();
+        }
+
+        public int GetBytes(Span<byte> buffer)
+        {
+            return this.reader.GetBytes(buffer);
+        }
+
+        public IEnumerable<SymbolToken> GetTypeAnnotations()
+        {
+            return this.reader.GetTypeAnnotations();
+        }
+
+        public byte[] NewByteArray()
+        {
+            return this.reader.NewByteArray();
+        }
+
+        public int GetLobByteSize()
+        {
+            return this.reader.GetLobByteSize();
+        }
+
+        private dynamic GetIonValue()
+        {
+            switch (this.CurrentType)
+            {
+                case IonType.Bool:
+                    return this.BoolValue();
+                case IonType.Blob:
+                    return this.NewByteArray();
+                case IonType.Clob:
+                    return this.NewByteArray();
+                case IonType.Decimal:
+                    return this.DecimalValue();
+                case IonType.Float:
+                    return this.DoubleValue();
+                case IonType.Int:
+                    return this.BigIntegerValue();
+                case IonType.String:
+                    return this.StringValue();
+                case IonType.Symbol:
+                    return this.SymbolValue();
+                case IonType.Timestamp:
+                    return this.TimestampValue();
+                default:
+                    throw new SystemException("Unexpected type '" + this.CurrentType + "'");
```

Comment:
i'm seeing some [general guidance](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/using-standard-exception-types?view=netframework-4.8) about not throwing `SystemException`.  perhaps there's a more appropriate alternative?

Reply:
Done

Reply:
see separate comments that propose exception types to use